[MIG] endpoint_auth_api_key: Migrate to version 16.0

sonhd91 opened the pull request:
Depended on unmerged:  `auth_api_key_group`

simahawk committed:
Add endpoint_auth_api_key

OCA-git-bot committed:
[ADD] icon.png

simahawk committed:
endpoint_auth_api_key: add btn to manage groups

simahawk committed:
endpoint_auth_api_key: show group mgmt btn if needed

OCA-git-bot committed:
endpoint_auth_api_key 14.0.1.1.0

simahawk committed:
Misc fix of authorship name

OCA-git-bot committed:
endpoint_auth_api_key 14.0.1.1.1

sebalix committed:
[FIX] Fix access right issues when using actions

OCA-git-bot committed:
endpoint_auth_api_key 14.0.1.1.2

simahawk committed:
endpoint_auth_api_key: move to OCA/web-api

simahawk committed:
endpoint_auth_api_key: adapt to new sync behavior

Commit:
[MIG] endpoint_auth_api_key: Migrate to version 16.0

nilshamerlinck commented on endpoint_auth_api_key/models/endpoint_mixin.py:
```diff
@@ -20,9 +20,9 @@ def _selection_auth_type(self):
         return super()._selection_auth_type() + [("api_key", "API key")]
 
     def _validate_request(self, request):
-        super()._validate_request(request)
```
careful with this type of change, by doing that you are changing the checks order
if you need a return to satisfy pre-commit, just add a final return

nilshamerlinck commented on endpoint_auth_api_key/tests/test_endpoint.py:
```diff
@@ -13,11 +13,11 @@
 class TestEndpoint(CommonEndpoint):
     @classmethod
     def _setup_records(cls):
-        super()._setup_records()
```
same here

sonhd91 commented:
Branch updated, please checked again

thienvh332 commented:
LGTM

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (43 of 43 strings)

Translation: web-api-16.0/web-api-16.0-webservice
Translate-URL: https://translation.odoo-community.org/projects/web-api-16-0/web-api-16-0-webservice/it/

JordiMForgeFlow committed:
[IMP] webservice: multi-company

OCA-git-bot committed:
Merge PR #32 into 16.0

Signed-off-by simahawk

Commit:
[UPD] Update webservice.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: web-api-16.0/web-api-16.0-webservice
Translate-URL: https://translation.odoo-community.org/projects/web-api-16-0/web-api-16-0-webservice/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (44 of 44 strings)

Translation: web-api-16.0/web-api-16.0-webservice
Translate-URL: https://translation.odoo-community.org/projects/web-api-16-0/web-api-16-0-webservice/it/
